src/custom/state/swap/hooks.ts

```diff
@@ -121,3 +133,106 @@ export function useDerivedSwapInfo(): DerivedSwapInfo {
     v1Trade: undefined
   }
 }
+
+// export function parseCurrencyFromURLParameter(urlParam: any): string {
+export function parseCurrencyFromURLParameter(urlParam?: string | string[] | ParsedQs | ParsedQs[]): string {
+  if (typeof urlParam === 'string' && urlParam?.toUpperCase() === 'ETH') return 'ETH'
+
+  const validTokenAddress = isAddress(urlParam)
+
+  if (typeof urlParam === 'string' && validTokenAddress) {
+    return validTokenAddress
+  } else {
+    // return empty token
+    return ''
+  }
+}
+
+export function queryParametersToSwapState(parsedQs: ParsedQs): SwapState {
+  let inputCurrency = parseCurrencyFromURLParameter(parsedQs.inputCurrency)
+  let outputCurrency = parseCurrencyFromURLParameter(parsedQs.outputCurrency)
+  if (inputCurrency === outputCurrency) {
+    if (typeof parsedQs.outputCurrency === 'string') {
+      inputCurrency = ''
+    } else {
+      outputCurrency = ''
+    }
+  }
+
+  const recipient = validatedRecipient(parsedQs.recipient)
+
+  return {
+    [Field.INPUT]: {
+      currencyId: inputCurrency
+    },
+    [Field.OUTPUT]: {
+      currencyId: outputCurrency
+    },
+    typedValue: parseTokenAmountURLParameter(parsedQs.exactAmount),
+    independentField: parseIndependentFieldURLParameter(parsedQs.exactField),
+    recipient
+  }
+}
+
+export function useReplaceSwapState() {
+  const dispatch = useDispatch()
+  return useCallback(
+    (newState: {
+      field: Field
+      typedValue: string
+      inputCurrencyId?: string | undefined
+      outputCurrencyId?: string | undefined
+      recipient: string | null
+    }) => dispatch(replaceSwapState(newState)),
+    [dispatch]
+  )
+}
+
+type DefaultFromUrlSearch = { inputCurrencyId: string | undefined; outputCurrencyId: string | undefined } | undefined
```

Comment:
All this stuff is copy paste,

I believe u didn't create a Mod in this case, becase `src/state/swap/hooks.ts`existed already, and had some copy paste. 

I won't stop the approve, but only if we promise a refactor, where we create the Mod version of this hook, and only override the parts we need to change.

Could you open an issue for this?

Reply:
done inside #205